src/components/FullscreenModal/FullscreenModal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { isUndefined, noop } from 'ramda-adjunct';
+
+import { getColor, getDepth } from '../../utils/helpers';
+import ModalHeader from './Header/Header';
+import ModalFooter from './Footer/Footer';
+import { FullscreenModalLayouts } from './FullscreenModal.enums';
+import { ColumnConfigMap, FullscreenModalProps } from './FullscreenModal.types';
+import { Col, Container, Row } from '../layout';
+
+const BaseModal = styled.div`
+  width: 100vw;
+  height: 100vh;
+  position: fixed;
+  top: 0;
+  left: 0;
+  overflow-y: auto;
+  background-color: ${getColor('graphite5H')};
+  z-index: ${getDepth('modal')};
+`;
+
+const columnConfigMap: ColumnConfigMap = {
+  [FullscreenModalLayouts.single6]: {
+    header: [6, 3],
+    sidebar: [0, 0],
+    content: [6, 3],
+  },
+  [FullscreenModalLayouts.single8]: {
+    header: [8, 2],
+    sidebar: [0, 0],
+    content: [8, 2],
+  },
+  [FullscreenModalLayouts.sidebar46]: {
+    header: [10, 1],
+    sidebar: [4, 1],
+    content: [6, 0],
+  },
+  [FullscreenModalLayouts.sidebar48]: {
+    header: [12, 0],
+    sidebar: [4, 0],
+    content: [8, 0],
+  },
+};
+
+const FullscreenModal: React.FC<FullscreenModalProps> = ({
+  layout = FullscreenModalLayouts.single6,
+  header,
+  content,
+  sidebar,
+  footer,
+  onClose = noop,
+}) => {
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  const {
+    header: headerConfig,
```

Comment:
readability suggestion - you could use `header: [headerCols, headerOffset]` etc instead of indexes